environment/src/main/java/com/sequenceiq/environment/environment/flow/creation/handler/ResourceGroupCreationHandler.java

```diff
@@ -0,0 +1,166 @@
+package com.sequenceiq.environment.environment.flow.creation.handler;
+
+import static com.sequenceiq.cloudbreak.common.mappable.CloudPlatform.AZURE;
+import static com.sequenceiq.environment.environment.flow.creation.event.EnvCreationHandlerSelectors.RESOURCEGROUP_CREATE_EVENT;
+import static com.sequenceiq.environment.environment.flow.creation.event.EnvCreationStateSelectors.START_NETWORK_CREATION_EVENT;
+import static com.sequenceiq.environment.parameters.dao.domain.ResourceGroupCreation.CREATE_NEW;
+
+import java.util.Optional;
+
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.cloud.azure.client.AzureClient;
+import com.sequenceiq.cloudbreak.cloud.azure.client.AzureClientService;
+import com.sequenceiq.cloudbreak.cloud.exception.CloudConnectorException;
+import com.sequenceiq.cloudbreak.cloud.init.CloudPlatformConnectors;
+import com.sequenceiq.cloudbreak.cloud.model.CloudCredential;
+import com.sequenceiq.cloudbreak.common.service.Clock;
+import com.sequenceiq.cloudbreak.exception.BadRequestException;
+import com.sequenceiq.cloudbreak.featureswitch.AzureSingleResourceGroupFeatureSwitch;
+import com.sequenceiq.cloudbreak.tag.CostTagging;
+import com.sequenceiq.environment.credential.v1.converter.CredentialToCloudCredentialConverter;
+import com.sequenceiq.environment.environment.domain.Environment;
+import com.sequenceiq.environment.environment.dto.EnvironmentDto;
+import com.sequenceiq.environment.environment.flow.creation.event.EnvCreationEvent;
+import com.sequenceiq.environment.environment.flow.creation.event.EnvCreationFailureEvent;
+import com.sequenceiq.environment.environment.service.EnvironmentService;
+import com.sequenceiq.environment.parameters.dto.AzureParametersDto;
+import com.sequenceiq.environment.parameters.dto.AzureResourceGroupDto;
+import com.sequenceiq.environment.parameters.dto.ParametersDto;
+import com.sequenceiq.environment.parameters.service.ParametersService;
+import com.sequenceiq.flow.reactor.api.event.EventSender;
+import com.sequenceiq.flow.reactor.api.handler.EventSenderAwareHandler;
+
+import reactor.bus.Event;
+import reactor.bus.EventBus;
+
+@Component
+public class ResourceGroupCreationHandler extends EventSenderAwareHandler<EnvironmentDto> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ResourceGroupCreationHandler.class);
+
+    private CloudPlatformConnectors cloudPlatformConnectors;
+
+    private AzureClientService azureClientService;
```

Comment:
I don't like the idea of adding direct Azure dependency (or any specific cloud provider dependency) in full generic Env creation flow steps.

Reply:
Can you name this step instead InfrastructureCreationHandler and move every cloud provider specific logic behind the cloud-api? And use those APIs here?

Reply:
I will do it as I see that also in the case of S3Guard this approach was taken and accept that the team likes it this way. However, I would like just note that this is somewhat against polymorphism, as we try to pretend something to be generic when in reality it isn't (resourceGroup is a pure azure concept). I would have factored it behind an interface if at least one another implementation (AWS or MOCK) to that flow step would be needed.

Reply:
If you name the step Infrastructure Creation it immediately becomes generic. It's a different question that for AWS this time there's nothing to do with it. But you know what features and requirements will there be in 3 months? 6 months? What if the need emerges to do some additional work on AWS during Env creation? We add a new AWS specific step? It would bloat the flow unnecessarily even more. Flows meant to be generic, not just in the Env service, but everywhere in CB, and cloud provider specific jobs should be hidden behind the cloud-api. Otherwise this whole software wouldn't work. This is a good legacy we have and we should guard it.

Reply:
Yes. One thought regarding the name: the whole cloudbreak is about infrastructure creation :), so the name is not really descriptive. And it is exactly this point, coming up with a good name that shows me that too much generics is premature now. 
Coming to the point what requirements will come in 3 months: well, I would be rather doing a refactor when we see what's needed (i.e. in 3 or 6 months time) then do a very generic solution with a suboptimal name that will need to be renamed anyway.
I agree that the flow is generic.

Reply:
I looked up S3GuardTable deletion. There the flow step is called that way and not masqueraded to be something generic, with good purpose: it is something that exists in the AWS world. Once there will be an azure parallel I am sure a heavy refactoring will come. So I will keep the name.

Reply:
I have not hard bias pro or contra, but I am merging the PR to make progress with the feature.  
@attilapalfi92 pls open a follow-up if you'd like to change this.